torchaudio/io/_playback.py

```diff
@@ -0,0 +1,55 @@
+import torch
+import torchaudio
+from torchaudio.io import StreamWriter
+from typing import Optional
+
+
+def play_audio(
+    waveform: torch.Tensor,
+    sample_rate: Optional[float],
+    device: Optional[str] = "audiotoolbox",
+    channels_first: Optional[bool] = True,
+    ) -> None:
+    """Play audio through specified or available output device
+
+    Args:
+        src: Tensor of the audio to play. Expected shape: (channels, time)
+            if channels_first is True, (time, channels) otherwise
+        sample_rate: Sample rate of the input audio
+        device: Output device. If None, use available device
+        channels_first: Whether the first dimension corresponds to
+            number of channels or not
+    """
+    dtype = waveform.dtype
+    try:
+        format = {
+            torch.uint8: "u8",
+            torch.int16: "s16",
+            torch.int32: "s32",
+            torch.int64: "s64",
+            torch.float32: "flt",
+            torch.float64: "dbl"
+        }[dtype]
+    except KeyError:
+        raise ValueError(f"Unsupported dtype: {dtype}")
```

Comment:
I agree with the above. Do you have a suggestion on where to place the dictionary with the formats? Somewhere where it can be easily re-used by other methods or test if necessary. I can place it in the top of this file for now (below the imports).

Reply:
I am indifferent.

It is attempting to put it in module-level, but that means the dictionary is created for all the users who do not use this function. It's kinda waste of memory, though it's very small. This function does not have to be super fast, so in my opinion, we don't gain much from putting it in module-level.

I'lld let you decide.